packages/venia-concept/src/components/App/errorNotifications.js

```diff
@@ -1,74 +1,50 @@
-import React, { Component } from 'react';
+import { useCallback, useEffect } from 'react';
 import { arrayOf, object, func, shape, string } from 'prop-types';
-import classify from 'src/classify';
-import { Notification, NotificationStack } from 'src/components/Notifications';
-import defaultClasses from './errorNotifications.css';
+import AlertCircleIcon from 'react-feather/dist/icons/alert-circle';
+import { useToastActions } from '@magento/peregrine';
 
 const dismissers = new WeakMap();
 
-class ErrorNotifications extends Component {
-    static propTypes = {
-        classes: shape({
-            debuginfo: string
-        }).isRequired,
-        errors: arrayOf(
-            shape({
-                error: object.isRequired,
-                id: string.isRequired,
-                loc: string
-            })
-        ),
-        onDismissError: func.isRequired
-    };
-
-    componentDidMount() {
-        window.scrollTo(0, 0);
-    }
-
-    dismissNotificationOnClick(e, dismiss) {
-        e.preventDefault();
-        e.stopPropagation();
-        dismiss();
-    }
+const ErrorNotifications = props => {
+    const { onDismissError } = props;
+    const { addToast } = useToastActions();
 
     // Memoize dismisser funcs to reduce re-renders from func identity change.
-    getErrorDismisser(error) {
-        const { onDismissError } = this.props;
-        return dismissers.has(error)
-            ? dismissers.get(error)
-            : dismissers.set(error, () => onDismissError(error)).get(error);
-    }
+    const getErrorDismisser = useCallback(
+        error => {
+            return dismissers.has(error)
+                ? dismissers.get(error)
+                : dismissers.set(error, () => onDismissError(error)).get(error);
+        },
+        [onDismissError]
+    );
+
+    return props.errors.map(({ error, id, loc }) => {
+        useEffect(() => {
+            // TODO: Without the enclosing useEffect this endless loops. Why?
```

Comment:
I don't understand why emitting the toast here with `addToast` within `<App />` could result in an endless loop. The only state/DOM that is updated should be in the top level `<ToastContainer />`. Is it because this all lives within `<ToastContextProvider>`? See [here](https://github.com/magento-research/pwa-studio/pull/1218/files#diff-1078ac6bf3201692a3f2202512f5b03dR41).

Reply:
Yea, I'd rather not enclose the `addToast` call with the hook but if I call it without `useEffect` the app infinitely loops and I don't know why.